tests/shells/test_bashshell.py

```diff
@@ -1,5 +1,135 @@
+import logging
+import os
+import pytest
+from crl.interactivesessions.InteractiveSession import InteractiveSession
+from crl.interactivesessions.shells.bashshell import (
+    BashShell,
+    BashShellTypeError)
+from crl.interactivesessions.shells.remotemodules.compatibility import to_bytes
+from .mockspawn import MockSpawn
+from .interpreter import (
+    HandlerBase,
+    HandlerNotMatching,
+    Response)
+
+
 __copyright__ = 'Copyright (C) 2019, Nokia'
 
 
+LOGGER = logging.getLogger(__name__)
+
+
+class SlowBannerHandler(HandlerBase):
+
+    _responses = [Response.create(b'Banner line 1'),
+                  Response.create(b'Banner line 2', latency=1)]
+
+    def get_responses(self, line):
+        path = line.split()[0]
+        if os.path.split(path)[-1] != b'bash':
+            raise HandlerNotMatching
+
+        self._state.prompt = self.initial_prompt
+        return self.responses
+
+    @property
+    def initial_prompt(self):
+        return b'initial-prompt'
+
+    @property
+    def responses(self):
+        for r in self._responses:
+            yield Response.create(r.response + b'\n', r.latency)
+
+        yield Response.create(self.initial_prompt)
+
```

Comment:
Minor refactoring: possibly would be more readable in case this full response is directly written to `_responses` instead of to the separate property.

Reply:
Yes, I agree.